### PR TITLE
Electron tray notifications badge

### PR DESCRIPTION
**Pull request by t3chguy**
for #3554 based on @lalomartins' fork

### CODE REVIEW

**Comment by t3chguy** [comment]
@dbkr 
1. its more to stop my IDE bullying me, if it sees an eslint file it starts complaining...a lot
2. I shall look into it
3. I don't see any reference to `electron/img` in that script.

**Comment by dbkr** [comment]
Another way of doing this would be to do it the same way as we do the favicon on web by taking the image favico.js spits out and setting that as the tray image, somehow (the tray takes a NativeImage as well as a path). This would mean not only would you not have to faff around putting red dots on icons but you'd also get the actual count.

**Comment by dbkr** [comment]
re: 3. indeed, there wasn't, which was a mistake, but I've fixed it now you've pointed it out. :)

**Comment by t3chguy** [comment]
@dbkr I personally do prefer dynamically generating it with the number, that way I can use the circle(N) for Windows taskbar badging which (at least through electron) only supports an overlay to the base image (does not generate the badge for us)

File: electron/src/tray.js [hunk 1]
```diff
@@ -47,21 +46,31 @@ exports.create = function (win, config) {
...
 
     trayIcon = new Tray(config.icon_path);
     trayIcon.setToolTip(config.brand);
     trayIcon.setContextMenu(contextMenu);
     trayIcon.on('click', toggleWin);
+
+    ipcMain.on('set_badge', function(event, count) {
+        if (count) {
+            trayIcon.setImage(config.icon_path_unread);
+            trayIcon.setToolTip(`${config.brand} — ${count} unread`);
```
Comment (dbkr) [inline electron/src/tray.js]: This text should probably match (and ideally be generated by the same code as) https://github.com/matrix-org/matrix-react-sdk/blob/master/src/components/structures/MatrixChat.js#L1105

**Comment by t3chguy** [comment]
superseded by #3909

**Pull request closed without merging**